### PR TITLE
State Rent change B placeholder EIP - net transaction counter

### PR DESCRIPTION
**Pull request by AlexeyAkhunov**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-X.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your Github username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: EIPS/eip-2031.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+---
+eip: 2031
+title: State Rent B - Net transaction counter
+author: Alexey Akhunov (@AlexeyAkhunov)
+discussions-to: https://ethereum-magicians.org/t/eip-2031-net-transaction-counter-change-b-from-state-rent-v3-proposal/3283
+status: Draft
+type: Standards Track
+category: Core
+created: 2019-05-15
+---
```
Comment (axic) [inline EIPS/eip-2031.md]: Need to add `requires: 2029`.

File: EIPS/eip-2031.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+non-contract accounts. This way of populating nonce ensures replay protection for accounts that were evicted and then brought back by sending ether to them.
+
+## Motivation
+<!--The motivation is critical for EIPs that want to change the Ethereum protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the EIP solves. EIP submissions without sufficient motivation may be rejected outright.-->
+Ethereum currently does not have a special place in the state for tracking number of transactions.
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+A new field, with the location 0, is added to the state counter contract. It will eventually contain `txCount`, the total number of transactions processed up until that point.
```
Comment (axic) [inline EIPS/eip-2031.md]: Can you explain what "a new field with location 0" means?

File: EIPS/eip-2031.md (hunk 1 above)
Comment (AlexeyAkhunov) [reply EIPS/eip-2031.md]: Thank you, done

File: EIPS/eip-2031.md (hunk 2 above)
Comment (AlexeyAkhunov) [reply EIPS/eip-2031.md]: Thanks, done

File: EIPS/eip-2031.md (hunk 1 above)
Comment (AlexeyAkhunov) [reply EIPS/eip-2031.md]: Thought it fails the check now, because it assumes that 2029 needs to be merged

Comment (axic) [reply EIPS/eip-2031.md]: Yes, we should merge 2029 first.

File: EIPS/eip-2031.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+in the State counter contract. This counter will be used to populate the nonces of newly created
+non-contract accounts. This way of populating nonce ensures replay protection for accounts that were evicted and then brought back by sending ether to them.
+
+## Motivation
+<!--The motivation is critical for EIPs that want to change the Ethereum protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the EIP solves. EIP submissions without sufficient motivation may be rejected outright.-->
+Ethereum currently does not have a special place in the state for tracking number of transactions.
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+A new field, with the location 0 (that means it resides in the storage slot 0 in the state counter contract, and can
```
Comment (axic) [inline EIPS/eip-2031.md]: I'm still not sure what "storage slot 0" means? Is that a 32 bytes value of all zeroes?

**Comment by axic** [comment]
@AlexeyAkhunov fixed some typos, but that one question still remains.

File: EIPS/eip-2031.md (hunk 3 above)
Comment (AlexeyAkhunov) [reply EIPS/eip-2031.md]: Yes, I added clarification for that, thank you!

**Pull request merged**